client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -179,9 +179,13 @@ const siteSetupFlow: Flow = {
 						);
 					}
 
-					// Add Launchpad to selected intents in General Onboarding
-					if ( isLaunchpadIntent( intent ) && typeof siteId === 'number' ) {
-						pendingActions.push( saveSiteSettings( siteId, { launchpad_screen: 'full' } ) );
+					// Update Launchpad option based on site intent
+					if ( siteId ) {
+						pendingActions.push(
+							saveSiteSettings( siteId, {
+								launchpad_screen: isLaunchpadIntent( intent ) ? 'full' : null,
+							} )
+						);
 					}
 
 					Promise.all( pendingActions ).then( () => window.location.assign( to ) );
```

Comment:
`NB:` Do we need to be catching exceptions here? Wondering what we do in the event that site settings fail to be toggled properly cc @Addison-Stavlo

Reply:
it looks like we were previously ensuring `siteId` was a number `&& typeof siteId === 'number'`. Should we still be doing that as well?

Reply:
@jeyip I've updated the PR to add back the `typeof` check but I'm not sure that was what you were referring to..